src/Microsoft.IdentityModel.Tokens/SecurityKey.cs

```diff
@@ -37,11 +37,23 @@ namespace Microsoft.IdentityModel.Tokens
     public abstract class SecurityKey
     {
         private CryptoProviderFactory _cryptoProviderFactory;
+        private readonly Lazy<string> _internalId;
 
         internal SecurityKey(SecurityKey key)
         {
             _cryptoProviderFactory = key._cryptoProviderFactory;
             KeyId = key.KeyId;
+            _internalId = new Lazy<string>(() =>
+            {
+                try
+                {
+                    return Base64UrlEncoder.Encode(ComputeJwkThumbprint());
```

Comment:
>ComputeJwkThumbprint() [](start = 51, length = 22)

consider moving out? Assuming this is the place where there could be an expected exception? Do you have expected exceptions from encoding?

Reply:
If we had a method: CanCreateJwkThumbprint(), then if that returns false, return set to string.Empty, we  would not need to throw in known cases.

Reply:
That's a good point. I'll add a public virtual method that returns false by default.

Reply:
Addressed with https://github.com/AzureAD/azure-activedirectory-identitymodel-extensions-for-dotnet/pull/1346/commits/0f00581f0a75b4972b983006eda9aad40e728414.